front/app/modules/commercial/insights/admin/containers/Insights/Details/Network/index.tsx

```diff
@@ -77,18 +90,57 @@ const StyledMessage = styled.h4`
 `;
 
 const Network = ({
-  params: { viewId },
   intl: { formatMessage, formatDate },
-  location: { query, pathname },
-}: WithRouterProps & InjectedIntlProps) => {
+}: InjectedIntlProps) => {
   const [initialRender, setInitialRender] = useState(true);
   const [height, setHeight] = useState(0);
   const [width, setWidth] = useState(0);
   const [zoomLevel, setZoomLevel] = useState(0);
+  const [hoverNode, setHoverNode] = useState<Node | undefined>();
+  const [pointerPosition, setPointerPosition] = useState([0, 0]);
+  const [hiddenNodes, setHiddenNodes] = useState<string[]>([]);
+  const [graphInitialized, setGraphInitialized] = useState<boolean>(false);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const { viewId } = useParams();
 
   const networkRef = useRef<ForceGraphMethods>();
   const { loading, network } = useNetwork(viewId);
   const view = useInsightsView(viewId);
+  const tooltipRef = document.getElementsByClassName('graph-tooltip')[0];
+
+  const setPointerEvent = (e) => setPointerPosition([e.offsetX, e.offsetY]);
+  const appendSearchParams = (params) =>
+    setSearchParams({
+      ...Object.fromEntries(searchParams.entries()),
+      ...params,
+    });
+
+  useEffect(() => {
+    const hidden_keywords = searchParams.get('hidden_keywords');
+    if (graphInitialized) {
+      if (!isNilOrError(network) && hidden_keywords) {
+        const {
+          data: {
+            attributes: { nodes },
+          },
+        } = network;
+        setHiddenNodes(
+          nodes
+            .map(({ id }: Node) => id)
+            .filter((nodeId) => hidden_keywords.includes(nodeId))
+        );
+      }
+
+      const canvasElement = document.getElementsByTagName('canvas')[0];
+      if (canvasElement) {
+        canvasElement.addEventListener('pointermove', setPointerEvent);
+        return () =>
+          canvasElement &&
+          canvasElement.removeEventListener('pointermove', setPointerEvent);
+      }
+    }
+    return;
+  }, [graphInitialized]);
```

Comment:
This doesn't seem to be the only dependency for this useEffect, could you add the missing ones?

Reply:
i have removed the `hiddenNodes` initialization all togheter